File: aphantasia_server/Aphant.Core.Dto/Concept.cs
namespace Aphant.Core.Dto;

public class ConceptLight
{
    public required string Tag { get; set; }

    public required string Color { get; set; }
}

public class Concept
{
    public required string Tag { get; set; }

    public required string Color { get; set; }

    public int FollowersCount { get; set; }

    public List<ThoughtNode> Thoughts { get; set; } = [];
}

public class ConceptGraphNode
{
    public required string Tag { get; set; }
    public required string Color { get; set; }
    public int ThoughtCount { get; set; }
}

public class ConceptGraph
{
    public List<ConceptGraphNode> Nodes { get; set; } = [];
}

File: aphantasia_web/src/model/dto/concept.ts
import type { ThoughtNode } from "./thought";

export interface Concept {
    tag: string;
    color: string;

    thoughts: ThoughtNode[];
}

export interface ConceptLight {
    tag: string;
    color: string;
}

export interface ConceptGraphNode {
    tag: string;
    color: string;
    thoughtCount: number;
}

export interface ConceptGraph {
    nodes: ConceptGraphNode[];
}